src/merge_sort.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+
+def merge_sort(a_list):
+    """Use merge sort to sort the list."""
+    if len(a_list) <= 1:
+        return a_list
+    left = []
+    right = []
+    for i in range(len(a_list)):
+        if i % 2 != 0:
+            left.append(a_list[i])
+        else:
+            right.append(a_list[i])
+    left = merge_sort(left)
+    right = merge_sort(right)
+    return _merge(left, right)
+
+
+def _merge(left, right):
+    """Takes two lists and returns single sorted list."""
+    new_list = []
+    while left and right:
+        if left[0] < right[0]:
+            new_list.append(left[0])
+            left = left[1:]
```

Comment:
This approach will work, but will be more expensive than necessary because creating a slice of a list will use an additional O(slice size) time and space.

The canonical merge sort implementation uses two separate index pointers for the left and right lists. You could also create two different generators which you only advance using `next()` when necessary.